core/services/chainlink/application.go

```diff
@@ -303,6 +306,28 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 			chainSet,
 			globalLogger,
 		)
+		// TODO [relay]: add flag for this experimental job type?
+		globalLogger.Debug("Off-chain reporting v2-relay enabled")
+		// TODO [relay]: move relayers creation outside OCR2 context (relayers will be multi-protocol)
+		relayers := relay.Relayers{
+			// TODO [relay]: support Ethereum as a relay
+			// TODO [relay]: solana OCR2 keys ('ocr2key.KeyBundle' is Ethereum specific)
+			relay.TypeSolana: solana.NewRelayer(keyStore.OCR2()),
+		}
```

Comment:
is there a way where the list of all the relays (and corresponding switch cases) only exist in one place (or file)?

it would be easier to add relays that way...something i remembered is how many different places the external initiator repo had code for switching the various networks. and it was always a bit difficult to track each place that needed an addition to the switch/case statements for a new network

Reply:
Please check out the latest implementation. There is now a master relayer, a delegate abstracting all available implementations (Ethereum & mocked Solana, for now).